lib/Gateway.js

```diff
@@ -771,43 +771,65 @@ function setDiscoveryValue (payload, prop, node) {
  * @param {*} cfg the discovery configuration
  * @param {import('../types').Z2MNode} node the node
  * @param {import('../types').Z2MValueId} valueId the valueId
- * @returns {any} a configuration if the rgb dimmer is added, false otherwise
+ * @returns {any} a configuration if the rgb dimmer is added
  */
-function addRgbDimmer (cfg, node, valueId) {
-  const rgbID = `51-${valueId.endpoint || 0}-hexColor`
-  const rgb = node.values[rgbID]
-  if (rgb) {
-    cfg = copy(cfg)
-    cfg.discovery_payload.rgb_state_topic = this.mqtt.getTopic(
-      this.valueTopic(node, rgb)
-    )
-    cfg.discovery_payload.rgb_command_topic =
-      cfg.discovery_payload.rgb_state_topic + '/set'
-    cfg.values = [rgbID]
-
-    const whiteID = `51-${valueId.endpoint || 0}-currentColor-0`
-    const whiteValue = node.values[whiteID]
-
-    if (whiteValue) {
-      const whiteIDWrite = `51-${valueId.endpoint || 0}-targetColor-0`
-      const whiteValueWrite = node.values[whiteIDWrite]
-      cfg.discovery_payload.white_value_state_topic = this.mqtt.getTopic(
-        this.valueTopic(node, whiteValue)
-      )
-      cfg.discovery_payload.white_value_scale = whiteValue.max
-      cfg.discovery_payload.white_value_template = '{{ value_json.value }}'
-      cfg.discovery_payload.white_value_command_topic = this.mqtt.getTopic(
-        this.valueTopic(node, whiteValueWrite),
-        true
-      )
 
-      cfg.values.push(whiteID, whiteIDWrite)
-    }
+function addRgbColorSwitch (cfg, node, valueId) {
+  cfg.discovery_payload.rgb_state_topic = this.mqtt.getTopic(
+    this.valueTopic(node, valueId)
+  )
+  cfg.discovery_payload.rgb_command_topic =
+    cfg.discovery_payload.rgb_state_topic + '/set'
+
+  // The following part of code, checks if ML or Binary works. If one exists the other
+  let brightnessValue = false
+  let switchValue = false
+  if (node.values[`38-${valueId.endpoint}-currentValue`]) {
+    brightnessValue = `38-${valueId.endpoint}-currentValue`
+  } else if (node.values[`38-${valueId.endpoint + 1}-currentValue`]) {
```

Comment:
Is this really needed? Why `+1`?

Reply:
I changed it a bit
fibaro uses endpoint 0 for hex but endpoint 1 for the 38 CC
new change is handling it better + comment

Remember still working on it! :) thanks for pointing out!